kinto/core/resource/__init__.py

```diff
@@ -245,6 +245,12 @@ def collection_get(self):
         self._raise_412_if_modified()
 
         headers = self.request.response.headers
+
+        if getattr(self.context, "forced_empty_list", False):
+            # Return an empty list.
+            headers['Total-Records'] = encode_header('0')
+            return self.postprocess([])
+
```

Comment:
Yes it is what I did in the first commit but it makes a query to the storage while we don't want to make one in that case.